include/solvers/gls_sharp_navier_stokes.h

```diff
@@ -57,94 +57,164 @@ class GLSSharpNavierStokesSolver : public GLSNavierStokesSolver<dim>
-
-  // clear all the line of dof even if the dof is not owned but it is ghost
-  void
-  clear_line_in_matrix(const typename DoFHandler<dim>::active_cell_iterator &cell, unsigned int dof_index);
-
-
-  // Modified version of assemble_matrix_and_rhs to include the presence of
-  // extra steps.
-  void
-  assemble_matrix_and_rhs(
-    const Parameters::SimulationControl::TimeSteppingMethod
-      time_stepping_method) override;
-
-  void
-  assemble_rhs(const Parameters::SimulationControl::TimeSteppingMethod
-                 time_stepping_method) override;
-
   /**
-   * Members
+  * @brief Calculate the coefficients required for BDF integration of order n.
+  * The coefficients are determined through a recursion algorithm.
+  * The algorithm is taken from : Hay, Alexander, et al. "hp-Adaptive time
+   * integration based on the BDF for viscous flows." Journal of Computational
+   * Physics 291 (2015): 151-176.
+   *
+   * @param order The order of the BDF method. The BDF method of order n requires n+1 arrays
+   *
+   * @param time_steps a vector containing all the time steps. The time steps should be in reverse order.
+   * For example, if the method is a BDF2, it uses three values for the time : (t)
+   * (t-dt_1) and (t-dt_2). Thus the time step vector should contain dt_1 and
+   * dt_2.
    */
+
+    /**
+    * @brief
+    * Map the vertex index to the cell that includes that vertex.
+    * This map is used to find all the cell close to a specific vertex.
+    */
+    void
+    vertices_cell_mapping();
+
+    /**
+    * @brief
+    * Defines the particle structure and value based on the parameter file.
+    */
+    void
+    define_particles();
+
+    /**
+    * @brief
+    * Evaluate the forces applied to each of the IB particles.
+    */
+    void
+    force_on_ib();
+
+    /**
+    * @brief
+    * Modify the system matrix to impose IB condition using the sharp_edge approach.
+    * The detail of this approach are presented in :
+     * L. Barbeau, S. Étienne, C. Béguin & B. Blais,
+     * «Development of a high-order continuous Galerkin sharp-interface immersed boundary method and
+     * its application to incompressible flow problems,» Computers & Fluids, 2020,
+     * in press, ref. CAF-D-20-00773
+    */
+    void
+    sharp_edge();
+
+    /**
+    * @brief
+    * Write in  the output file the forces , velocity , position of each of the
+    * particles.
+    */
+    void
+    write_force_ib();
+
+    /**
+    * @brief
+    * Write in  the output file the forces , velocity , position of each of the
+    * particles.
+    */
+    void
+    integrate_particles();
+
+    /**
+    * @brief
+    * Store the solution of the particles dynamics parameters for integration.
+    * Defines the table to store the history of each of the particles.
+    */
+    void
+    finish_time_step_particles();
+
+    /**
+    * @brief
+    * Evaluate the L2 error on the computational domain if an analytical solutio
+    * is given.
```

Comment:
Can you add detail as to how this is different than for when you don't have particles? Is this a seperate routine now because you need to take into account the particles, or could you use the regular one from Lethe? If you cannot, you need to explain why here.